### PR TITLE
PIM-10123: Merge 5.0 into master

### PR DESCRIPTION
**Pull request by anaisbaunelemaire**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

**Review by nmarniesse — APPROVED** [review]

**Comment by anaisbaunelemaire** [comment]
warm

File: upgrades/schema/Version_5_0_20211018134301_migrate_longtext_to_json.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+namespace Pim\Upgrade\Schema;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+/**
+ * Catalogs created in 3.0 (and older ones) have columns created with longtext type. See PIM-10040.
+ * To be consistent with new catalogs we migrate them in json.
+ */
+final class Version_5_0_20211018134301_migrate_longtext_to_json extends AbstractMigration
```
Comment (rfauglas) [inline upgrades/schema/Version_5_0_20211018134301_migrate_longtext_to_json.php:14]: Are we sure we keep it like that? a new migration marked as  5.0 on master?

**Review by rfauglas — CHANGES_REQUESTED** [review]

File: upgrades/schema/Version_5_0_20211018134301_migrate_longtext_to_json.php (hunk 1 above)
Comment (anaisbaunelemaire) [reply upgrades/schema/Version_5_0_20211018134301_migrate_longtext_to_json.php:14]: Good remark, I'll investigate!

Comment (nmarniesse) [reply upgrades/schema/Version_5_0_20211018134301_migrate_longtext_to_json.php:14]: Yes we need it. It's not a mandatory migration, there is no bug with/without it. So the migration is authorized in this case.   

So what is the purpose of this migration you may ask? It helps to standardize the database of old/new instance. In order to have the same column type. So in the future if we use a SQL json function in a query, we will not have error on old catalog

Comment (anaisbaunelemaire) [reply upgrades/schema/Version_5_0_20211018134301_migrate_longtext_to_json.php:14]: @nmarniesse, I think that @rfauglas was talking about the class name, marked as 5.0. Isn't it?

**Review by rfauglas — APPROVED** [review]

File: src/Akeneo/Pim/Enrichment/Component/Product/Connector/FlatTranslator/AttributeValue/MultiSelectTranslator.php [hunk 2]
```diff
@@ -38,7 +37,7 @@ public function translate(string $attributeCode, array $properties, array $value
 
         $result = [];
         foreach ($values as $valueIndex => $value) {
-            if (empty($value)) {
+            if (null === $value || '' === $value) {
```
[the remark this answers is not in the document]
Comment (anaisbaunelemaire) [reply src/Akeneo/Pim/Enrichment/Component/Product/Connector/FlatTranslator/AttributeValue/MultiSelectTranslator.php:40]: This modification was made as part of a bug fix:
https://github.com/akeneo/pim-community-dev/commit/cb430833688b6614f5b4004a51b156815143650a#diff-30028da9774e88fe7067c1cefeb4f1cb1995ed22ef26c983cee84a5a9a6be9b1

**Comment by anaisbaunelemaire** [comment]
warm

**Pull request merged**